scripts/bump_major_version.py

```diff
@@ -0,0 +1,335 @@
+# For example, to do all of the admin tasks around bumping the version of
+# CockroachDB from 20.1 to 20.2, and then surfacing the places where additional
+# edits may be needed, run:
+#
+# $ ./bump_major_version.py -d /path/to/docs -f 20.1 -t 20.2
+
+parser = argparse.ArgumentParser(
+    description="do admin tasks related to version bumps"
+)
+
+parser.add_argument("-f", "--from-version",
+                    required=True,
+                    help="previous version of CockroachDB")
+
+parser.add_argument("-t", "--to-version",
+                    required=True,
+                    help="next version of CockroachDB")
+
+parser.add_argument("-d", "--docs-directory",
+                    default=os.environ.get("COCKROACH_DOCS_REPO"),
+                    help="directory where you store 'cockroachdb/docs'")
+
+args = parser.parse_args()
+
+from_version = args.from_version
+to_version = args.to_version
+docs_directory = args.docs_directory
+
+# Every file we touch that has content in it will also be stored here.  At the
+# end we will iterate over the lines of each file and do some simple-minded
+# transformations (mostly regexes to check and bump version numbers, and remove
+# "New in vXX.1" tags).
+files_to_check = {}
+
+
+def version_string_name(version):
+    # Int -> String
+    return 'v' + str(version)
+
+
+def make_new_version_folder(version):
+    # Int -> IO!
+    new_directory = os.path.join(docs_directory, version_string_name(version))
+    if not os.path.exists(new_directory):
+        os.mkdir(new_directory)
+
+
+def copy_files(from_version, to_version):
+    # Int Int -> IO! State!
+    src_dir = os.path.join(docs_directory,
+                           version_string_name(from_version))
+    dest_dir = os.path.join(docs_directory, version_string_name(to_version))
+    if os.path.exists(dest_dir):
+        shutil.rmtree(dest_dir)
+    shutil.copytree(src_dir, dest_dir)
+    marktree(dest_dir)
+
+
+def marktree(path):
+    # Path -> State!
+
+    # Recursively add the files in this subtree to our list of files to check
+    # later for mentions of the previous version number.
+    for dirpath, dirnames, filenames in os.walk(path):
+        for file in filenames:
+            f = os.path.join(dirpath, file)
+            if f not in files_to_check:
+                files_to_check[f] = 1
+
+
+def copy_includes(from_version, to_version):
+    # Int Int -> IO! State!
+    src_dir = os.path.join(docs_directory, '_includes',
+                           version_string_name(from_version))
+    dest_dir = os.path.join(docs_directory, '_includes',
+                            version_string_name(to_version))
+
+    if os.path.exists(dest_dir):
+        shutil.rmtree(dest_dir)
+    shutil.copytree(src_dir, dest_dir)
+
+    # Traverse down the directory containing the "include files", adding each
+    # to our list of files to check later for references to the old version.
+    marktree(dest_dir)
+
+
+def copy_images(from_version, to_version):
```

Comment:
Since `copy_files`, `copy_includes`, and `copy_images` are essentially performing the same steps, maybe you can condense them into one function with a third parameter passed along to modify `src_dir` and `dest_dir` as necessary.

Reply:
Done. Added a flag to `copy_files` that determines whether it's "regular" docs files, included files, or images that need to be dealt with.  Not sure if the arg-passing style is correct cuz I'm not really a Python-knower.  PTAL.